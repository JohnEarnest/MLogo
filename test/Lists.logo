# Lists
#
# test primitive list operations

print size  []
print size  [A B C]

print first    []
print first    [A B C]
print last     []
print last     [A B C]
print butfirst []
print butfirst [A B C]
print butlast  []
print butlast  [A B C]

print flatten  []
print flatten  [A [B [C []]]]
print flatten  [[A B][C D] E]

print item 0 [A B C]
print item 2 [A B C]
print item 5 [A B C]

print fput 'Z []
print fput 'Z [A B]
print lput 'Z []
print lput 'Z [A B]

print join [A B][C]
print join [][A B C]
print join [A B C][]

print member 'food [dog food in cans]
print member 'pork [acceptable kosher foods]
