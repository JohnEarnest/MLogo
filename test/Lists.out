0
3
[]
A
[]
C
[]
[B C]
[]
[A B]
[]
[A B C]
[A B C D E]
A
C
[]
['Z]
['Z A B]
['Z]
[A B 'Z]
[A B C]
[A B C]
[A B C]
[food in cans]
[]
